url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/StandAloneSimulation.java

```diff
@@ -0,0 +1,527 @@
+  		double co2 = Integer.parseInt((sfp.propSimulation.getProperty("co2") != null ? sfp.propSimulation.getProperty("co2").trim() : "0")); 	
+  		String setup = sfp.propScenario.getProperty("setup");
+	    SimulationParams simParams = new SimulationParams(responseType, scenarioName, locationInfo, numOfDays, 
+	    		startDateDay,  startDateMonth, startDateYear, mcruns, co2, setup);
+	    
+	   
+			
+	    if (setup.equals("dynamic"))
+	    {
+	    		if (sfp.demographics.isEmpty())
+	    		{
+					try {
+						throw new Exception("ERROR: Demographic data have to be provided for dynamic scenarios.");
+					} catch (Exception e) {
+						e.printStackTrace();
+						System.exit(0);
+					}
+	    		}
+	    		else
+	    		{
+			    String dName = sfp.demographics.getProperty("name") != null ? sfp.demographics.getProperty("name") : "";
+			    String dType = sfp.demographics.getProperty("type") != null ? sfp.demographics.getProperty("type") : "";
+			    String dDescription = sfp.demographics.getProperty("description") != null ? sfp.demographics.getProperty("description") : "";
+			    int number_of_entities = Integer.parseInt((sfp.demographics.getProperty("number_of_entities") != null ? sfp.demographics.getProperty("number_of_entities").trim() : "1")); 
+			    String participation_probs_installations = sfp.demographics.getProperty("participation_probs_installations") != null ? sfp.demographics.getProperty("participation_probs_installations") : "";
+			    String participation_probs_persons = sfp.demographics.getProperty("participation_probs_persons") != null ? sfp.demographics.getProperty("participation_probs_persons") : "";
+			    String participation_probs_apps = sfp.demographics.getProperty("participation_probs_apps") != null ? sfp.demographics.getProperty("participation_probs_apps") : "";
+			    
+			    this.demographics = new DemographicData(dName, dDescription, dType, number_of_entities, 
+			    		parseProbabilities(participation_probs_installations), parseProbabilities(participation_probs_persons), parseProbabilities(participation_probs_apps));
+	    		}
+	    }
+	    
+ 		PricingPolicy pricPolicy = new PricingPolicy();
+ 		if (!sfp.propPricing.isEmpty() && !sfp.propPricing.getProperty("type").equals("None"))
+ 		{
+ 			String pricingType = sfp.propPricing.getProperty("type");		
+	  		int billingCycle = Integer.parseInt((sfp.propPricing.getProperty("billingCycle") != null ? sfp.propPricing.getProperty("billingCycle").trim() : "0")); 
+	  		double fixedCharge = Integer.parseInt((sfp.propPricing.getProperty("fixedCharge") != null ? sfp.propPricing.getProperty("fixedCharge").trim() : "0")); 
+	  		PricingPolicy.Builder builderPP = new PricingPolicy.Builder(pricingType, fixedCharge, billingCycle);	
+	  		try {
+				pricPolicy = constructPricingPolicy(pricingType, builderPP);
+			} catch (Exception e) {
+				e.printStackTrace();
+			}
+ 		} 			
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 113:124, 127:138.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.